data/transactions/logic/box.go

```diff
@@ -165,9 +169,59 @@ func MakeBoxKey(appIdx basics.AppIndex, name string) string {
 	   The "bx:" prefix is so that the kvstore might be usable for things
 	   besides boxes.
 	*/
-	key := make([]byte, 3 /* bx: */ +8 /* appIdx, big-endian */ +len(name))
-	copy(key, "bx:")
-	binary.BigEndian.PutUint64(key[3:], uint64(appIdx))
-	copy(key[11:], name)
+	key := make([]byte, boxSuffixIndex + len(name))
+	copy(key, boxPrefix)
+	binary.BigEndian.PutUint64(key[boxPrefixLength:], uint64(appIdx))
+	copy(key[boxSuffixIndex:], name)
 	return string(key)
 }
+
+// GetAppAndNameFromKey extracts an appid and box name from a string that was created by MakeBoxKey()
+func GetAppAndNameFromKey(key string) (basics.AppIndex, string, error) {
```

Comment:
This is useful for box ingestion in indexer, but in theory, it could live there, not here. Open to suggestions.

Reply:
I'd like something simpler like `SplitBoxKey` but not intending to bike shed